Src/BootCamp.Chapter/Inventory.cs

```diff
@@ -1,26 +1,47 @@
-﻿namespace BootCamp.Chapter
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection.Metadata.Ecma335;
+
+namespace BootCamp.Chapter
 {
     public class Inventory
     {
-        private Item[] _items;
-        public Item[] GetItems()
-        {
-            return new Item[0];
-        }
+        private const int defaultInventorySize = 10;
+
+        public List<Item> Items { get; private set; }
+        public int Size { get; private set; }
+        public int RemainingSpace { get; set; }
 
         public Inventory()
         {
-            _items = new Item[0];
+            Size = defaultInventorySize;
+            RemainingSpace = defaultInventorySize;
+            Items = new List<Item>(defaultInventorySize);
         }
 
-        public Item[] GetItems(string name)
+        public Item[] GetItem(string name)
         {
+            if (string.IsNullOrWhiteSpace(name)) throw new ArgumentException($"{nameof(name)} cannot be null or empty.");
+
+            if (RemainingSpace == Size) return new Item[0];
+
+            foreach (var item in Items)
+            {
+                if (item.Name == name)
+                {
+                    return new Item[] { item };
+                }
+            }
+
             return new Item[0];
         }
 
         public void AddItem(Item item)
         {
+            if (RemainingSpace == 0) return;
 
+            Items.Add(item ?? throw new ArgumentNullException($"{nameof(item)} cannot be null."));
```

Comment:
Consider consistent behavior with your other methods here. One situation that you might see unexpected behavior - 

- Remaining space is 0. 
- `AddItem(null)` is called. 
- Method returns, making no change. 

Compared to 

- RemainingSpace > 0. 
- `AddItem(null)` is called. 
- Throws an `ArgumentNullException`. 

I think this method should *always* throw an `ArgumentNullException` if `item is null`.

Reply:
You're right on the fact that it should always throw an **ArgumentNullException** if **item is null**.

The 1st behaviour, however, doesn't seem to be right.

- Remaining space is 0. -> returns immediately and never calls AddItem(null).

So in my head, it seemed more logical to check whether the inventory has any available spaces left or not and only worry about the item if there is at least one. Though you might be right, and maybe it would be better to check for null before all.

This option, however, results in a bit less code as I'd have to call something like:
if (item is null) throw new ArgumentNullException(); before the the rest of the method's body.

I'm not sure though, maybe in the long run it would be beneficial to have the null-check first and that would remain consistent with other methods that do null-checking as the first step.